test/modules/OS/POSIX/pipe.chpl

```diff
@@ -0,0 +1,27 @@
+use OS.POSIX;
+
+var fildes:2*c_int;
+const ret = pipe(c_ptrTo(fildes));
+writeln(ret == 0);
+if ret == 0 {
+  // The pipe was created properly.  Here we create two tasks.  One
+  // writes to the write end and the other reads from the read end.
```

Comment:
Wouldn't this not work if the two tasks were scheduled on the same thread?

Reply:
I think at least this one will work, because of the sync var synchronization. But `select.chpl`, below, doesn't have any synchronization.

Reply:
This is addressed by https://github.com/chapel-lang/chapel/pull/19632/commits/eabbd841b89137ea9b7ade4655241a50e85b5bc7.